components/common/src/test/java/com/hotels/styx/api/ResponseEventListenerTest.java

```diff
@@ -167,5 +168,19 @@ public void firesResponseContentError() {
         assertTrue(responseError.get() instanceof RuntimeException);
     }
 
+    @Test
+    public void firesErrorWhenResponseCompletesWithoutHeaders() {
+        TestSubscriber<LiveHttpResponse> testSubscriber = new TestSubscriber<>();
+        Observable<LiveHttpResponse> publisher = Observable.empty();
 
+        ResponseEventListener.from(publisher)
+                .whenResponseError(cause -> responseError.set(cause))
+                .apply()
+                .toBlocking()
+                .subscribe(testSubscriber);
+
+        testSubscriber.awaitTerminalEvent();
+        assertEquals(testSubscriber.getOnCompletedEvents().size(), 1);
+        assertTrue(responseError.get() instanceof RuntimeException);
```

Comment:
We could use `throw responseError.get();` in order to use the expectedException annotation.  What do you think?

Reply:
I would leave the `expectedException` annotation for situations where an exception is actually *thorwn*. But here the exception is merely emitted through the publisher instead of being thrown. Therefore I would leave the code as it is.

Reply:
TestNG also has an annotation that can check the message of the exception, but it we don't need to do that here, no problem.